File: sdk/cpp/examples/basic.cpp
/// \file
/// Basic ErrorBrain C++ SDK example

#include <errorbrain/errorbrain.h>
#include <iostream>
#include <map>

int main() {
    std::cout << "ErrorBrain C++ SDK - Basic Example" << std::endl;
    std::cout << "==================================" << std::endl << std::endl;

    try {
        ErrorBrain::ErrorBrainClient client("http://localhost:8000");

        // Health check
        std::cout << "Checking API health..." << std::endl;
        auto health = client.health_check();
        std::cout << "✅ API is healthy" << std::endl;
        std::cout << "   Status: " << health.status() << std::endl;
        std::cout << "   LLM Configured: " << (health.llm_configured() ? "yes" : "no")
                  << std::endl << std::endl;

        // Create error report
        std::map<std::string, std::string> metadata{
            {"user_id", "12345"},
            {"request_id", "abc-def"},
        };

        ErrorBrain::ErrorReport report("cpp", "billing-service",
                                       "Database connection timeout");
        report.with_traceback("thread panicked at 'connection timeout'\n   at "
                              "db/connection.cpp:42:10")
            .with_tags({"prod", "database"})
            .with_metadata(metadata);

        // Send error
        std::cout << "Sending error report..." << std::endl;
        auto response = client.send_error(report);
        std::cout << "✅ Error sent successfully" << std::endl;
        std::cout << "   ID: " << response.id() << std::endl;
        std::cout << "   Explanation: " << response.explanation() << std::endl;
        if (response.saved_path()) {
            std::cout << "   Saved to: " << *response.saved_path() << std::endl;
        }

    } catch (const ErrorBrain::ErrorBrainException& e) {
        std::cerr << "❌ Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

File: sdk/rust/examples/basic.rs
//! Rust/Cargo example: Basic error reporting

use errorbrain_sdk::{ErrorBrainClient, ErrorReport};
use std::collections::HashMap;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("ErrorBrain Rust SDK - Basic Example");
    println!("==================================\n");

    let client = ErrorBrainClient::new("http://localhost:8000".to_string());

    // Health check
    match client.health_check().await {
        Ok(health) => {
            println!("✅ API is healthy");
            println!("   Status: {}", health.status);
            println!("   LLM Configured: {}\n", health.llm_configured);
        }
        Err(e) => {
            eprintln!("❌ Health check failed: {}\n", e);
            return Err(e.into());
        }
    }

    // Create error report
    let mut metadata = HashMap::new();
    metadata.insert("user_id".to_string(), "12345".to_string());
    metadata.insert("request_id".to_string(), "abc-def".to_string());

    let report = ErrorReport::new("rust", "billing-service", "Database connection timeout")
        .with_traceback(
            "thread 'main' panicked at 'connection timeout'\n   at db/connection.rs:42:10"
                .to_string(),
        )
        .with_tags(vec!["prod".to_string(), "database".to_string()])
        .with_metadata(metadata);

    // Send error
    match client.send_error(&report).await {
        Ok(response) => {
            println!("✅ Error sent successfully");
            println!("   ID: {}", response.id);
            println!("   Explanation: {}", response.explanation);
            if let Some(path) = response.saved_path {
                println!("   Saved to: {}", path);
            }
        }
        Err(e) => {
            eprintln!("❌ Failed to send error: {}", e);
            return Err(e.into());
        }
    }

    Ok(())
}
